### DISCUSSION TITLE
Next15 cookies async

### DISCUSSION THREAD

**Discussion by olli-suoniemi** [amannn/next-intl#1604] [General]
Just in case someone is following the example from [here](https://github.com/amannn/next-intl/tree/main/examples/example-app-router-without-i18n-routing) AND using NextJS 15, the setting/getting of the cookies won't work

In Next.JS 15, the cookies is an async function. Check example from [documentation](https://nextjs.org/docs/app/api-reference/functions/cookies#examples )

I was doing

`await cookies().set(COOKIE_NAME, locale);`

and wondering why it doesn't work, but the correct syntax is:

`(await cookies()).set(COOKIE_NAME, locale);`

**Comment by username14415** [comment]
check this
```
const cookieStore = await cookies();
cookieStore.set(COOKIE_NAME, locale);
```
